api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go

```diff
@@ -179,339 +179,355 @@ func (i *MaterializeStageProvider) Process(ctx context.Context, mgrContext conte
 		namespace = "default"
 	}
 
-	mLog.DebugfCtx(ctx, "  P (Materialize Processor): masterialize %v in namespace %s", prefixedNames, namespace)
+	mLog.DebugfCtx(ctx, "  P (Materialize Processor): materialize %v in namespace %s", prefixedNames, namespace)
 
-	var catalogs []model.CatalogState
-	catalogs, err = i.ApiClient.GetCatalogs(ctx, namespace, i.Config.User, i.Config.Password)
-
-	if err != nil {
-		mLog.ErrorfCtx(ctx, "Failed to get catalogs: %s", err.Error())
+	catalogs := make(map[string]model.CatalogState)
+	errorMessage := "Failed to get all catalogs: "
+	for _, object := range prefixedNames {
+		object := api_utils.ConvertReferenceToObjectName(object)
+		catalog, err := i.ApiClient.GetCatalog(ctx, object, namespace, i.Config.User, i.Config.Password)
+		if err != nil {
+			errorMessage = fmt.Sprintf("%s %s(reason: %s).", errorMessage, object, err.Error())
+			continue
+		}
+		catalogs[object] = catalog
+	}
+	if len(catalogs) < len(prefixedNames) {
+		mLog.ErrorCtx(ctx, errorMessage)
 		providerOperationMetrics.ProviderOperationErrors(
 			materialize,
 			functionName,
 			metrics.ProcessOperation,
 			metrics.RunOperationType,
 			v1alpha2.CatalogsGetFailed.String(),
 		)
-		return outputs, false, err
+		return outputs, false, v1alpha2.NewCOAError(nil, errorMessage, v1alpha2.NotFound)
 	}
-	creationCount := 0
-	for _, catalog := range catalogs {
-		for _, object := range prefixedNames {
-			object := api_utils.ConvertReferenceToObjectName(object)
-			if catalog.ObjectMeta.Name == object {
-				label_key := os.Getenv("LABEL_KEY")
-				label_value := os.Getenv("LABEL_VALUE")
-				annotation_name := os.Getenv("ANNOTATION_KEY")
-				if label_key != "" && label_value != "" {
-					// Check if metadata exists, if not create it
-					metadata, ok := catalog.Spec.Properties["metadata"].(map[string]interface{})
-					if !ok {
-						metadata = make(map[string]interface{})
-						catalog.Spec.Properties["metadata"] = metadata
-					}
 
-					// Check if labels exists within metadata, if not create it
-					labels, ok := metadata["labels"].(map[string]string)
-					if !ok {
-						labels = make(map[string]string)
-						metadata["labels"] = labels
-					}
+	createdObjectList := make(map[string]bool, 0)
+	for _, object := range prefixedNames {
```

Comment:
Is there any reason why you iterate over prefixedNames rather than catalogs map? And is it possible that exist==false in line 211?

Reply:
1. The `prefixNames` can be used to log the names of catalogs easier. (line 525)
2. `exist == false` cannot happen since we have already checked `len(catalogs) < len(prefixedNames)` (line 195)

Reply:
catalogs names can also be retrieved from catalog object